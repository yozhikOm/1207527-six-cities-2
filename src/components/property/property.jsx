import React from 'react';
import PropTypes from 'prop-types';

const Property = (props) => {
  return <section className="property">
    <div className="property__gallery-container container">
      <div className="property__gallery">
        <div className="property__image-wrapper">
          <img className="property__image" src="img/room.jpg" alt="Photo studio" />
        </div>
        <div className="property__image-wrapper">
          <img className="property__image" src="img/apartment-01.jpg" alt="Photo studio" />
        </div>
        <div className="property__image-wrapper">
          <img className="property__image" src="img/apartment-02.jpg" alt="Photo studio" />
        </div>
      </div>
    </div>
    <div className="property__container container">
      <div className="property__wrapper">
        <div className="property__mark">
          <span>Premium</span>
        </div>
        <div className="property__name-wrapper">
          <h1 className="property__name">{props.title}</h1>
          <button className="property__bookmark-button button" type="button">
            <span className="visually-hidden">To bookmarks</span>
          </button>
        </div>
      </div>
    </div>
    <div className="property__price">
      <b className="property__price-value">&euro;{props.price}</b>
      <span className="property__price-text">&nbsp;night</span>
    </div>
  </section>;
};

Property.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
};

Property.defaultProps = {
  title: `Some title`,
  type: `Room`,
  price: 88
};

export {Property};
