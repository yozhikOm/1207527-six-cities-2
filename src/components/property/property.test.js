import React from 'react';
import {Property} from './property.jsx';
import renderer from 'react-test-renderer';

it(`Property компонент рендерится корректно`, () => {
  const propertyComponent = renderer.create(<Property/>).toJSON();
  expect(propertyComponent).toMatchSnapshot();
});
