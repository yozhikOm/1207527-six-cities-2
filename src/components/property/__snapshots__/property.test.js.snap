// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property компонент рендерится корректно 1`] = `
<section
  className="property"
>
  <div
    className="property__gallery-container container"
  >
    <div
      className="property__gallery"
    >
      <div
        className="property__image-wrapper"
      >
        <img
          alt="Photo studio"
          className="property__image"
          src="img/room.jpg"
        />
      </div>
      <div
        className="property__image-wrapper"
      >
        <img
          alt="Photo studio"
          className="property__image"
          src="img/apartment-01.jpg"
        />
      </div>
      <div
        className="property__image-wrapper"
      >
        <img
          alt="Photo studio"
          className="property__image"
          src="img/apartment-02.jpg"
        />
      </div>
    </div>
  </div>
  <div
    className="property__container container"
  >
    <div
      className="property__wrapper"
    >
      <div
        className="property__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="property__name-wrapper"
      >
        <h1
          className="property__name"
        >
          Some title
        </h1>
        <button
          className="property__bookmark-button button"
          type="button"
        >
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="property__price"
  >
    <b
      className="property__price-value"
    >
      €
      88
    </b>
    <span
      className="property__price-text"
    >
       night
    </span>
  </div>
</section>
`;
